LightTheMaze.py

```diff
@@ -0,0 +1,259 @@
+                            self.win = True
+
+                #If up is pressed
+                if( pygame.key.get_pressed()[pygame.K_UP] != 0 ):
+                    #Check for collisions
+                    if self.currentMap.getCollision(self.pxpos,self.pypos-1):
+                        #If collided, add a point
+                        self.points = self.points + 1
+                    else:
+                        #else, move
+                        self.pypos = self.pypos - 1
+                        if self.currentMap.getEnd(self.pxpos, self.pypos):
+                            #If you ended the maze, win
+                            self.win = True
+
+                #If m is pressed
+                if( pygame.key.get_pressed()[pygame.K_m] !=0):
+                    #If you have matches
+                    if self.matches >= 1:
+                        #Set match flag to true, and start a timer
+                        self.matchislit = True
+                        self.start_ticks=pygame.time.get_ticks() #starter tick
+        #If you win the game
+        if self.win:
+            #Add in reset event
+            #If s is pressed
+            if( pygame.key.get_pressed()[pygame.K_s] !=0):
+                #Reset map
+                self.on_init()
+
+
+
+    def on_loop(self):
+        """
+        Method runs any logic or math that needs to be run in the game loop.
+
+        Primarily used for Match timer.
+        """
+
+        #If player has lit a match
+        if self.matchislit:
+
+            #Check timer since lighting match
+            seconds=(pygame.time.get_ticks()-self.start_ticks)/1000 #calculate how many seconds
+            self.matchtimer= 3 - seconds
+            #Print seconds left
+            self.matchtimerprint = self.font.render(str(self.matchtimer),1,(200,200,200))
+
+            #if match expired
+            if seconds>3:
+                #Decrement number of matches and set match flag to false.
+                self.matches = self.matches -1# if more than 10 seconds close the game
+                self.matchislit = False
+
+
+    def on_render(self):
+
+        #Creates Player and Match images
+        player = pygame.image.load("pictures/player.png")
+        matches = pygame.image.load("pictures/matchStick.png")
+
+        #blist black background to screen
+        self.screen.blit(self.background,(0,0))
+        #self.screen.blit(point,(470,0))
+        #Creat
+        self.pointsprint = self.font.render("Points: " + str(self.points),1,(200,200,200))
+        self.matchprint = self.font.render(": "+str(self.matches), 1, (200, 200, 200))
+
+        #If match is lit
+        if self.matchislit:
+        #    self.screen.blit(self.mapdrawing,(80,80))
+            #Display map and timer
+            self.screen.blit(self.matchtimerprint, (300,40))
+            self.screen.blit(self.currentMap.map_drawing,(80,80))
+
+        #If you win
```

Comment:
Comments need some work. Many of your comments are unnecessary (you don't need to tell us when you are assigning a value to a variable)